core/trino-main/src/main/java/io/trino/sql/planner/optimizations/HashGenerationOptimizer.java

```diff
@@ -926,27 +929,33 @@ public boolean canComputeWith(Set<Symbol> availableFields)
 
         private Expression getHashExpression(Session session, Metadata metadata, TypeProvider types)
         {
-            Expression hashExpression = new GenericLiteral(StandardTypes.BIGINT, Integer.toString(INITIAL_HASH_VALUE));
-            for (Symbol field : fields) {
-                hashExpression = getHashFunctionCall(session, hashExpression, field, metadata, types);
+            Expression hashExpression = hashCodeExpression(session, metadata, types, fields.get(0));
+            for (int i = 1; i < fields.size(); i++) {
+                hashExpression = getHashFunctionCall(session, hashExpression, fields.get(i), metadata, types);
             }
             return hashExpression;
         }
 
         private static Expression getHashFunctionCall(Session session, Expression previousHashValue, Symbol symbol, Metadata metadata, TypeProvider types)
         {
-            FunctionCall functionCall = FunctionCallBuilder.resolve(session, metadata)
-                    .setName(QualifiedName.of(HASH_CODE))
-                    .addArgument(types.get(symbol), symbol.toSymbolReference())
-                    .build();
+            Expression functionCall = hashCodeExpression(session, metadata, types, symbol);
 
             return FunctionCallBuilder.resolve(session, metadata)
                     .setName(QualifiedName.of("combine_hash"))
                     .addArgument(BIGINT, previousHashValue)
-                    .addArgument(BIGINT, orNullHashCode(functionCall))
+                    .addArgument(BIGINT, functionCall)
                     .build();
         }
 
+        private static Expression hashCodeExpression(Session session, Metadata metadata, TypeProvider types, Symbol symbol)
```

Comment:
Can we have a common function for this and the `hashCodeExpression` in the outer class ?
They're doing almost same thing